spyder/plugins/application/container.py

```diff
@@ -297,10 +311,25 @@ def _check_updates_ready(self):
                         "<code>conda install spyder={}</code><br><br>"
                     ).format(latest_release)
                 else:
-                    content = _(
-                        "Click <a href=\"{}\">this link</a> to "
-                        "download it.<br><br>"
-                    ).format(url_r)
+                    if os.name == 'nt' and is_pynsist():
+                        box.setStandardButtons(QMessageBox.Yes |
+                                               QMessageBox.No)
+                        content = _(
+                            "You want to download and install the latest "
+                            "version of spyder?<br><br>"
+                        )
+                        box.buttonClicked.connect(
+                            lambda button:
+                            self.application_update_status.start_installation()
+                            if button.text() in ('&Yes')
+                            else
+                            self.application_update_status.set_status_pending()
+                            )
```

Comment:
I think we could further simplify this by using the [`exec_` method](https://doc.qt.io/qt-5/qmessagebox.html#exec) instead of the `show` method on line 336. With that you can use the [`result` method](https://doc.qt.io/qt-5/qdialog.html#result) and compare the return value to the `QMessageBox.Yes` value. So this could be removed for something like the following at line 336:

```python
box.exec_()
if box.result() == QMessageBox.Yes:
    self.application_update_status.start_installation()

```